mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -37,45 +37,74 @@ use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
-    }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
-        let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
-            "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
-            }
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let platform_ticker = coin.platform_ticker();
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx.get_coins_to_disable(platform_ticker, &ticker).await;
+
+    let mut disabled_tokens_tickers = vec![];
+    let mut cancelled_orders = vec![];
+    for ticker in &coins_to_disable {
+        log!("disabling {ticker} coin");
+        let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+        if !swaps.is_empty() {
+            let err = json!({
+                "error": format!("There're active swaps using {}", ticker),
+                "swaps": swaps,
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+        let (cancelled, still_matching) = try_s!(
+            cancel_orders_by(&ctx, CancelBy::Coin {
+                ticker: ticker.to_string()
+            })
+            .await
+        );
+
+        if !still_matching.is_empty() {
+            let err = json!({
+                "error": format!("There're currently matching orders using {}", ticker),
+                "orders": {
+                    "matching": still_matching,
+                    "cancelled": &cancelled,
+                }
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+
+        try_s!(disable_coin_impl(&ctx, ticker).await);
+        // Abort all coin related futures on coin deactivation
+        if let Err(err) = coin.on_disabled() {
+            log!("Error aborting coin futures: {err}")
+        };
+
+        // Combine all orders to a single vector
+        cancelled_orders.extend(cancelled);
+        disabled_tokens_tickers.push(ticker);
     }
 
-    try_s!(disable_coin_impl(&ctx, &ticker).await);
     let res = json!({
         "result": {
-            "coin": ticker,
-            "cancelled_orders": cancelled,
+            "coins": disabled_tokens_tickers,
```

Comment:
@sergeyboyko0791 Do you think this change breaks backward compatibility for GUIs? If that's the case, it might be better to add a new field in the results for `disabled_tokens_tickers` vec instead and leave the coin field as is.

Reply:
I still haven't reviewed the changes yet, hope will do it today

Reply:
But yes, this is good to me
>  add a new field in the results for disabled_tokens_tickers vec instead and leave the coin field as is.